python_modules/dagster/dagster/_core/execution/plan/compute_generator.py

```diff
@@ -143,7 +143,11 @@ def _validate_multi_return(
     context: OpExecutionContext,
     result: Any,
     output_defs: Sequence[OutputDefinition],
-) -> None:
+) -> Any:
+    # narrow further on what conditions its valid?
+    if result is None and context.has_assets_def:
```

Comment:
would this also catch if an op has multiple `Outs` specified, but only returns a single None?

Reply:
The current gating is `has_assets_def` , so a plain op with multiple outs wont trigger this behavior. I am not sure thats right. 

Some other conditions to consider: 
* all outputs are `Nothing` - wont work in the base case since the defaults are `Any` . Can make the new "AssetNode" variant use `Nothing` then this would facilitate that. 
* all outputs are `Nothing` or `Any` - expands previous to cover the base case 
* unconditional - just make this new `None` becomes `(None, ...)` the always behavior.

Reply:
After thinking on it my current leading contender is

* interpret implicit or explicit `return None`  to mean `return (None, None, ...)` for any multiple output asset or op. Disallow when there are "optional" outputs to prevent control flow misinterpretation.